### PR TITLE
Return IActionResult from API actions

### PR DESCRIPTION
**Pull request by bfren**

### CODE REVIEW

File: src/Notes/Strasnote.Notes.Api/Controllers/NoteController.cs [hunk 1]
```diff
@@ -5,57 +5,126 @@
...
 		[HttpPost]
-		public Task<long> Create() =>
-			Is.AuthenticatedUser(
-				ctx,
-				userId => notes.CreateAsync(new()
-				{
-					UserId = userId
-				}),
-				() => 0
+		[SwaggerResponse(201, "The note was created.", typeof(long))]
```
Comment (cwn-dev) [inline src/Notes/Strasnote.Notes.Api/Controllers/NoteController.cs:42]: We can use [ProducesResponse] instead so we're not tied to Swagger/Swashbuckle

**Review by cwn-dev — COMMENTED** [review]
I don't really understand where this is looking at e.g. the UserId value on a note & ensuring the user is allowed to update that note?

File: src/Notes/Strasnote.Notes.Api/Controllers/NoteController.cs (hunk 1 above)
Comment (bfren) [reply src/Notes/Strasnote.Notes.Api/Controllers/NoteController.cs]: All done

**Comment by bfren** [comment]
> I don't really understand where this is looking at e.g. the UserId value on a note & ensuring the user is allowed to update that note?

Yes it passes the current UserId so if a user tries to update a note that isn't theirs, it fails.

**Pull request merged**